src/utils/ota/ota.go

```diff
@@ -2,80 +2,41 @@
 package ota
 
 import (
-	"bufio"
-	"fmt"
-	"log"
-	"os"
-
 	"github.com/blang/semver"
+	"github.com/pkg/errors"
 	"github.com/rhysd/go-github-selfupdate/selfupdate"
 )
 
-// DoSelfUpdate updates the app to the latest version
-func DoSelfUpdate() {
+// DoAutoUpdate updates the app to the latest version.
+func DoAutoUpdate() (updateFound bool, newVersion, changeLog string, err error) {
 	v, err := semver.ParseTolerant(Version)
 	if err != nil {
-		log.Println("Binary version validation failed:", err)
-
+		err = errors.Wrap(err, "binary version validation failed")
```

Comment:
Please use fmt.Errorf here, no need to import additional dependency just to wrap errors (even if it's faster - this is not a hot path)